app/services/rating.js

```diff
@@ -0,0 +1,50 @@
+const { Rating, sequelize } = require('../models');
+const logger = require('../logger');
+const { databaseError } = require('../errors');
+const { calculateUserScore } = require('../helpers/calculate-user-score');
+
+exports.createRating = async (rating, user) => {
+  try {
+    const transaction = await sequelize.transaction();
+    const { score } = rating;
+    const { score: userScore } = user;
+    const ratingInfo = await Rating.create(rating, { transaction });
+
+    await user.update({ score: (userScore + score) }, { transaction });
+    await transaction.commit();
+    return ratingInfo;
+  } catch (error) {
+    logger.error(error);
+    if (transaction.rollback) await transaction.rollback();
+    throw databaseError(error);
+  }
+};
+
+exports.updateRating = async (score, rating, user) => {
+  try {
+    const transaction = await sequelize.transaction();
+    const { score: currentScore } = rating;
+    const { score: currentUserScore } = user;
+    const newRatingScore = currentScore === score ? 0 : score;
```

Comment:
If the new score is the same a the current one, just do nothing and return. I think you are over complicating things here 😅